classes/form/CustomerForm.php

```diff
@@ -117,15 +117,16 @@ public function validate()
 
         // birthday is from input type text..., so we need to convert to a valid date
         $birthdayField = $this->getField('birthday');
-        if (!empty($birthdayField->getValue()) &&
-            Validate::isBirthDate($birthdayField->getValue(), Context::getContext()->language->date_format_lite)
-        ) {
-            $dateBuilt = DateTime::createFromFormat(
-                Context::getContext()->language->date_format_lite,
-                $birthdayField->getValue()
-            );
-            $birthdayField->setValue($dateBuilt->format('Y-m-d'));
-        }
+        if (!empty($birthdayField) {
```

Comment:
Maybe we should only allow this if `PS_CUSTOMER_BIRTHDATE` is disabled ?
Something like:
```suggestion
        $isBirthDateRequired = (bool) Configuration::get('PS_CUSTOMER_BIRTHDATE');
        if ($isBirthDateRequired || !empty($birthdayField)) {
```
Also a `)` was missing 😉

Reply:
Hello Matks,
Assume && instead of ||, as the later could still break the code.

I wasn't aware of the constants, although I went through the documentation. So yes, your suggestion is great in taking part of the handling away from the view (&& instead of ||), but still we need to have that field in memory, so I see needless redundancy, and some sort of validation is still carried on at the second half of the nested if statement:
```
    if (!empty($birthdayField->getValue()) &&
        Validate::isBirthDate($birthdayField->getValue(),  
Context::getContext()->language->date_format_lite)
        ) 
```
Assuming the birthdate validation is perfect at validating birthdate's

And I feel awful wasting Travis time.